src/index.ts

```diff
@@ -32,8 +33,26 @@ import { Chain } from './chainList'
 import Popup from './popup'
 import { ModalController } from './ui/modalController'
 
+function preLoadIframe(url: string, appId: string) {
+  try {
+    if (typeof document === 'undefined') return
+    const iframeLink = document.createElement('link')
+    iframeLink.href = `${url}/${appId}/login`
+    iframeLink.type = 'text/html'
+    iframeLink.rel = 'prefetch'
+    if (iframeLink.relList && iframeLink.relList.supports) {
+      if (iframeLink.relList.supports('prefetch')) {
+        document.head.appendChild(iframeLink)
+      }
+    }
+  } catch (error) {
+    console.warn(error)
+  }
+}
+
 class AuthProvider {
```

Comment:
class name is `Arcana Provider` and importing a fn with a same name, can we rename any one?

cc: @ajithranka

Reply:
@karthik-durai What?

Reply:
sorry my bad, ignore this.